Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -239,7 +241,10 @@ public final class PostgresConnection: @unchecked Sendable {
             promise: promise
         )
 
-        self.channel.write(HandlerTask.extendedQuery(context), promise: nil)
+        let writePromise = self.channel.eventLoop.makePromise(of: Void.self)
+        self.channel.write(HandlerTask.extendedQuery(context), promise: writePromise)
+        writePromise.futureResult.cascadeFailure(to: promise)
```

Comment:
Does it make sense to move those three lines into a private function?

Reply:
done